### PR TITLE
Add refcounts

### PR DESCRIPTION
**Pull request by peterkxie**
@zeeev and i wrote the functions to chunk up the refseq and create count tables, then add the ref count to the SVIdx. mainRefCounter is the main function to call

### CODE REVIEW

File: src/nibpkg/refmers.nim [hunk 1]
```diff
@@ -1,23 +1,65 @@
...
+        chrom_name: string
+        chrom_start: int
+        chrom_end: int
+    
+proc addRefCount*(refKmers: CountTableRef[uint64], svKmers: svIdx) =
+    for k, v in pairs(refKmers): 
+        if svKmers.hasKey(k):
+            svKmers[k].refCount += uint32(v)
+
+proc createChunk*(fai: Fai, chunk_size: int): seq[Chunk] =
```
Comment (brentp) [inline src/nibpkg/refmers.nim]: I would make this an iterator:


```Nim
iterator chunks*(fai:Fai, chunksize:int): Chunk =
  ...
     yield chunk
```

File: src/nibpkg/refmers.nim [hunk 2]
```diff
@@ -1,23 +1,65 @@
...
+    var chunkSeq = newSeq[Chunk]()
+    for i in 0..<fai.len:
+        let chrom_name = fai[i]
+        let chrom_len = fai.chrom_len(chrom_name)
+        for j in countup(0, chrom_len, chunk_size):
+            var chunk: Chunk
```
Comment (brentp) [inline src/nibpkg/refmers.nim]: here, you can replace following 5 lines with:
`yield Chunk(chrom_name:chrom_name, chrom_start:j, chrom_end: j + chunk_size)`

File: src/nibpkg/refmers.nim [hunk 3]
```diff
@@ -1,23 +1,65 @@
...
+proc countRefKmers*(input_fn: string, kmer_size: int=21, chunk_size: int=1000000) =
     ##Walk over reference sequences and count kmers
     var fai: Fai
     if not fai.open(input_fn):
         quit "couldn't open fasta"
+    
+    let chunks = createChunk(fai, chunk_size)
+    for i in chunks:
+        let chunkCount = countByChunk(fai, i, kmer_size)
+        echo(chunkCount)
```
Comment (edawson) [inline src/nibpkg/refmers.nim:43]: I assume this writes to stdout - could we write to a file-like object instead (optionally stdout or a filename)?

**Review by edawson — CHANGES_REQUESTED** [review]
Can we remove the refkmers binary?

**Review by edawson — APPROVED** [review]

**Review by pb-cdunn — APPROVED** [review]

**Comment by pb-cdunn** [comment]
This is good, Peter. I only changed it per suggestions.

**Pull request merged**